### PR TITLE
feat: support WMS on reearth/core

### PR DESCRIPTION
**Pull request by keiya01**
# Overview

## What I've done

I've implement support of wms. Also I've added raster appearance type.

## Screenshot

Screen Shot 2022-12-22 at 10 17 46

### CODE REVIEW

File: src/core/engines/Cesium/Feature/Raster/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  const { layers, zIndex, minimumLevel, maximumLevel, credit } = property ?? {};
+  const { viewer } = useCesium();
+  const [type, url] = useMemo((): [RasterAppearance["type"], string | undefined] => {
+    const data = extractSimpleLayerData(layer);
+    const type = layer?.raster?.type;
+    const url = layer?.raster?.url;
+    return [type ?? (data?.type as RasterAppearance["type"]), url ?? data?.url];
+  }, [layer]);
+
+  const imageryProvider = type ? imageryProviders[type] : undefined;
```
Comment (rot1024) [inline src/core/engines/Cesium/Feature/Raster/index.tsx:25]:
```suggestion
  const imageryProvider = type ? ImageryProviders[type] : undefined;
```

File: src/core/mantle/types/appearance.ts [hunk 2]
```diff
@@ -139,6 +140,16 @@ export type ResourceAppearance = {
   clampToGround?: boolean;
 };
 
+export type RasterAppearance = {
+  url?: string;
+  type?: "wms";
```
Comment (rot1024) [inline src/core/mantle/types/appearance.ts:145]:
```suggestion
```

Type should be handled on `data`. Type is not a "style" and "appearance" of raster.

File: src/core/mantle/types/appearance.ts [hunk 3]
```diff
@@ -139,6 +140,16 @@ export type ResourceAppearance = {
   clampToGround?: boolean;
 };
 
+export type RasterAppearance = {
+  url?: string;
```
Comment (rot1024) [inline src/core/mantle/types/appearance.ts:144]:
```suggestion
```

url should be handled on `data`. URL is not style or appearance of raster. (glTF has no geo info and it cannot be handled as a feature, but wms can produce features, so it's data)

File: src/core/engines/Cesium/Feature/Raster/index.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  layers: [
+    {
+      id: "l",
+      type: "simple",
+      data: {
+        type: "wms",
+        url: "",
+      },
+      raster: {
+        layers: [""],
```
Comment (rot1024) [inline src/core/engines/Cesium/Feature/Raster/index.stories.tsx:30]: empty url?

```suggestion
      data: {
        type: "wms",
        url: "https://gibs.earthdata.nasa.gov/wms/epsg3857/best/wms.cgi",
        layer: "IMERG_Precipitation_Rate"
      },
      raster: {},
```

Note: that data is open data so you can use it in storybook

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: src/core/engines/Cesium/Feature/Raster/index.tsx (hunk 1 above)
Comment (keiya01) [reply src/core/engines/Cesium/Feature/Raster/index.tsx]: I fixed it
[chore: rename](https://github.com/reearth/reearth-web/pull/387/commits/41c7093f58cd509e1e8b401340bf4c41504d176a)

File: src/core/mantle/types/appearance.ts (hunk 2 above)
Comment (keiya01) [reply src/core/mantle/types/appearance.ts]: I fixed it, also the comment below is fixed.
[fix: data type](https://github.com/reearth/reearth-web/pull/387/commits/3444ed9819ace9e93a84837595fd4d3ed59f52e3)

File: src/core/engines/Cesium/Feature/Raster/index.stories.tsx (hunk 4 above)
Comment (keiya01) [reply src/core/engines/Cesium/Feature/Raster/index.stories.tsx]: It is fixed
[test: fix storybook data](https://github.com/reearth/reearth-web/pull/387/commits/141346a1bd72b9e52dcb0ccd0f5b4d69b20fcef1)

File: src/core/mantle/types/appearance.ts [hunk 5]
```diff
@@ -139,6 +140,13 @@ export type ResourceAppearance = {
   clampToGround?: boolean;
 };
 
+export type RasterAppearance = {
+  zIndex?: number;
```
Comment (rot1024) [inline src/core/mantle/types/appearance.ts:144]:
```suggestion
```

Layer order can be z-index so it can be deleted

**Review by rot1024 — APPROVED** [review]

File: src/core/mantle/types/appearance.ts (hunk 5 above)
Comment (keiya01) [reply src/core/mantle/types/appearance.ts]: [fix: remove zindex](https://github.com/reearth/reearth-web/pull/387/commits/a272691bbad01b41f36abe836cdb3480091c0b06)

**Review by rot1024 — APPROVED** [review]

**Review by KaWaite — APPROVED** [review]

**Pull request merged**